### DISCUSSION TITLE
ESP 32 Meplhaa Script help

### DISCUSSION THREAD

**Discussion by Hanimaniac** [RavenSystem/esp-homekit-devices#2029] [General]
Hey Everyone,

i tried my first ESP32 Module with HAA 12 but at some point I don't come to an conclution.
Its an ESP 2-Channel Module (
https://a.aliexpress.com/_EJPDfvJ ) and I build this skript:

{"c":{"io":[[[16,17,0],2],[[2,0],6],[[5,13],6,0,1],[[12],7]],"l":0,"b":[{"g":2,"t":5}],"n":"Markise-Terrasse"},
"a":[{"t":45,"o":18,"c":14,"f":70,
"0":{"r":[{"g":17},{"g":16,"v":1}]},
"1":{"r":[{"g":17,"v":1},{"g":16}]},
"2":{"r":[{"g":17},{"g":16}]},
"3":{"r":[{"g":17},{"g":16,"i":0.2}]},
"4":{"r":[{"g":17,"i":0.2},{"g":16}]},
"f3":[{"g":5,"t":0}],
"f4":[{"g":13,"t":0}],
"es":[{"t":30,"g":[12],"b":[{"g":0}]},{"t":1}]}]}

It controls an awning with dimmable LED Lighting (and a dummy switch for future projects).

The Installation woked fine and when I reboot the esp32 module, I can find it in my Home App but can't add it.
After the reboot the esp Module always stays in setup mode, but the HAA Installer don't say anything like bad Meplhaa skript or something.

Edit 18.05.23
In the Setup Mode, it says pairings 0/40 if this is interresting. 

I hope someone ca help me.
Thanks Martin

**Comment by QDUNI** [comment]
Did you fix it ?

**Reply by Hanimaniac** [reply]
Hello,
I could't fix it yet. 
The same Script works on an esp8266 like a charm. 
Maybe it's somethimg wrong with flashing. I don't know it at the moment.

**Reply by QDUNI** [reply]
Well, You configure GPIO 0 to be input only at first (2) and directly after you try to configure it as output only... (6) maybe this causes your problem. I will have a look to the rest of the script tomorrow. 


"io":[[[16,17,0],2],[[2,0],6],[[5,13],6,0,1]]

Try changing it into: 

"io":[[[16,17],2],[[0,5,13],6,1]] 

For b use gpio 0

Also be aware in HAA 12.x
You should write 

"b": [{"g":0,"t":5}]

Like this 

"b": [[0,5]]

And 

"r":[{"g":17},{"g":16,"v":1}]

Like this 

"r":[[17],[16,1]]

**Reply by Hanimaniac** [reply]
Thanks for your Help. I will have a look.

**Reply by QDUNI** [reply]
Try This,

Maybe I messed up an action... with low or high. 

````
{"c":{"io":[[[16,17],2],[[5,13,0],6,1],[[12],7]],"l":0,"b":[[0,5]],"n":"Markise-Terrasse"},"a":[{"t":45,"o":18,"c":14,"f":70,"0":{"r":[[17],[16,1]]},"1":{"r":[[17,1],[16]]},"2":{"r":[[17],[16]]},"3":{"r":[[17],[16,0,0.2]]},"4":{"r":[[17,0,0.2],[16]]},"f3":[[5,0]],"f4":[[13,0]],"es":[{"t":30,"g":[12],"b":[[0]]},{"t":1}]}]}
